### PR TITLE
Add Philips RDM004 (HUE Wall Switch rev 004) to quirks

### PR DESCRIPTION
**Pull request by Tijgerd**
## Proposed change
Added the RDM004 to the quirks.
Credits to Regenbui (Wesley Westerveld):  [https://community.home-assistant.io/t/how-to-configure-the-philips-hue-wall-module-to-use-push-button-momentary-type-wall-switches-zha/451125/27](url)

## Additional information
The RDM004 was not usable in ZHA, the fix as proposed by "Regenbui" was working as expected!
Also fixed a minor 

## Checklist
- [x] The changes are tested and work correctly
- [ ] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

File: zhaquirks/philips/rdm004.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+
+class PhilipsBasicCluster(CustomCluster, Basic):
+    """Philips Basic cluster."""
+
+    attributes = Basic.attributes.copy()
+    attributes.update(
+        {
+            0x0031: ("philips", t.bitmap16, True),
+            0x0034: ("mode", t.enum8, True),
```
Comment (MattWestb) [inline zhaquirks/philips/rdm004.py]: Some spaces is after the comma that the tests dont like.
Put  in `            0x0034: ("mode", t.enum8, True),` and it shall working OK

**Comment by MattWestb** [comment]
Looks nice but one new error is detected:
```
zhaquirks/philips/rdm004.py:3:1: UP035 `typing.List` is deprecated, use `list` instead
Found 1 error.

Error: Process completed with exit code 1.
```
I dont knowing if its only tinging the command or its needs more work then im not one code warier.

**Comment by Tijgerd** [comment]
Give me a second :D will fix this!
But it was working in home assistant like a charm

**Comment by MattWestb** [comment]
All green !!!!
Only function code test implantation failing.

**Comment by Tijgerd** [comment]
I see this is also failing on the RDM001 (and will possibly also fail on the other quirks in the Philips folder.
Looking into this but I am not able to properly debug this myself at the moment..

**Comment by Tijgerd** [comment]
This I am unable to get the tests to work...
I am running the custom quirk for over a week now without any problem, it is just the test that I am unable to integrate..

File: tests/test_philips.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        },
+        "manufacturer": "Philips",
+        "model": "RDM001",
+        "class": "zhaquirks.philips.rdm001.PhilipsRDM001",
+    }
+
+    assert_signature_matches_quirk(
+        zhaquirks.philips.rdm001.PhilipsRDM001, signature
+    )
```
Comment (MattWestb) [inline tests/test_philips.py]: Its one space after the ) that making the test complaining.

Comment (MattWestb) [reply tests/test_philips.py]: The other:
```
+++ b/tests/test_philips.py
@@ -1,11 +1,7 @@
 """Tests for Philips RDM001 and RDM004 quirks."""
-import pytest
 
-from tests.common import ClusterListener
 
 import zhaquirks
-from zhaquirks.philips.rdm001 import PhilipsRDM001
-from zhaquirks.philips.rdm004 import PhilipsRDM004
 
 zhaquirks.setup()
```
Its  looks the test i not liking importing this function but i dont  knowing if its needed for the code or not then im not one code warier.
I think Puddly or TheJullian is knowing it bettter.

Comment (Tijgerd) [reply tests/test_philips.py]: nope, the space after the ) was not the problem apparently

Comment (MattWestb) [reply tests/test_philips.py]: The first error is now gone only the import section and the the strange last ones . . . so one step forewords and . . .

**Comment by TheJulianJES** [comment]
This should be implemented with the following PRs:
- https://github.com/zigpy/zha-device-handlers/pull/2340
- https://github.com/zigpy/zha-device-handlers/pull/3075
- https://github.com/zigpy/zha-device-handlers/pull/3543

Everything should be available in Home Assistant Core 2024.12. Some of it is already available in 2024.11.
@Tijgerd Still, thanks for the PR!

**Pull request closed without merging**